tools/dump_tool/dump_tool.c

```diff
@@ -290,13 +294,78 @@ static int ramdump_recv(int dev_fd)
 				}
 			}
 			printf("]\n");
+			fflush(stdout);
 			fclose(bin_fp);
 		}
 	}
 	return 0;
 
 }
 
+static int filedump_recv(int dev_fd)
+{
+	int ret;
+	uint32_t smartfs_addr;
+	uint32_t smartfs_size;
+	char fs_bin[BINFILE_NAME_SIZE] = { '\0' };
+	FILE *fs_fp;
+	char buf;
+	uint32_t count = 0;
+
+	/* Static for now - ARTIK, to be made dynamic */
+	smartfs_addr = 0x04620000;
+	smartfs_size = 1024000;
+
+	printf("\n=========================================================================\n");
+	printf("Filesystem start address = %x, filesystem size = %d\n", smartfs_addr, smartfs_size);
+	printf("=========================================================================\n");
+
+	/* Create filesystem dump binary file name */
+	snprintf(fs_bin, BINFILE_NAME_SIZE, "fsdump_0x%08x_0x%08x.bin", smartfs_addr, (smartfs_addr + smartfs_size));
+
+	printf("\nReceiving file system dump.....\n");
+
+	fs_fp = fopen(fs_bin, "w");
+	if (fs_fp == NULL) {
+		printf("%s create failed\n", fs_bin);
+		return -1;
+	}
+
+	printf("[>");
+	fflush(stdout);
+
+	/* Dump data of Memory REGIONx  */
```

Comment:
I feel you can create a separate API to dump data so that its not repeated here. You can pass size, file pointer & dev_fd as arguments.

Reply:
Updated